core/integration/bookkeeping/plugin.go

```diff
@@ -117,7 +118,22 @@ func (p *Plugin) CallStack(data interface{}) (stack map[string]interface{}) {
 	varStack := call.VarStack
 	envId, ok := varStack["environment_id"]
 	if !ok {
-		log.Error("cannot acquire environment ID")
+		err := errors.New("cannot acquire environment ID")
```

Comment:
Similar comment to the other PR: should these parse failures return an error? I cannot judge. Otherwise looks good